File: matrices/search-a-2d-matrix-ii/search-a-2d-matrix-ii.java
class Solution {
  public boolean searchMatrix(int[][] matrix, int target) {

    if (matrix == null || matrix.length == 0 || matrix[0].length == 0) {
      return false;
    }

    int h = matrix.length;
    int w = matrix[0].length;

    int i = 0;
    int j = w - 1;

    while (i < h && j >= 0) {
      if (matrix[i][j] == target) {
        return true;
      } else if (matrix[i][j] > target) {
        j--;
      } else {
        i++;
      }
    }
    return false;
  }
}

File: matrices/search-a-2d-matrix-ii/search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
function searchMatrix(matrix, target) {
  if (!matrix || !matrix.length || !matrix[0].length) {
    return false;
  }

  const h = matrix.length;
  const w = matrix[0].length;

  let i = 0;
  let j = w - 1;

  while (i < h && j >= 0) {
    if (matrix[i][j] === target) {
      return true;
    } else if (matrix[i][j] > target) {
      j--;
    } else {
      i++;
    }
  }
  return false;
}
